svgdigitizer/electrochemistry/cv.py

```diff
@@ -18,111 +18,166 @@
+unit_typos = {'uA / cm2': ['uA / cm2', 'uA / cm²', 'µA / cm²', 'µA cm⁻²', 'uA cm-2', 'uA / cm2'],
+              'A / cm2': ['A / cm2', 'A cm⁻²', 'A cm-2', 'A / cm2'],
+              'A': ['A', 'ampere', 'amps', 'amp'],
+              'mV': ['milliV', 'millivolt', 'milivolt', 'miliv', 'mV'],
+              'V': ['V', 'v', 'Volt', 'volt'],
+              'V / s': ['V s-1', 'V / s'],
+              'mV / s': ['mV / s', 'mV s-1', 'mV/s']}
 
 
 class CV():
-    def __init__(self, metadata, svgplot):
+    def __init__(self, svgplot, metadata=None):
         """
         metadata: dict
         """
         self.svgplot = svgplot
-        self.metadata = metadata
-        self.svgplot.create_df()
+        self.metadata = metadata or {}
 
-        # TODO: These labels should either be extracted from the svg or yaml file
-        self.xlabel = 'U'
+    @property
+    @cache
+    def axis_properties(self):
+        return {'x': {'dimension': 'U',
+                      'unit': 'V'},
+                'y': {'dimension': 'j' if 'm2' in str(self.get_axis_unit('y')) else 'I',
+                      'unit': 'A / m2' if 'm2' in str(self.get_axis_unit('y')) else 'A'}}
 
-        # TODO: should be j if both current and normalized to are given in the yaml file
-        self.ylabel = 'I'
+    def get_axis_unit(self, axis):
+        r'''
+        Replaces the units derived from the svg file into astropy units.
+        '''
+        unit = self.svgplot.units[axis]
 
-        # TODO: All the rest in the init is presumably not necessary
-        self.description = self.metadata['figure description']
+        return self.get_correct_unit(unit)
 
-        self.xunit = self.description['potential scale']['unit']
+    @classmethod
+    def get_correct_unit(cls, unit):
+        for correct_unit, typos in unit_typos.items():
+            if unit in typos:
+                return u.Unit(correct_unit)
 
-        self.yunit = self.description['current']['unit']
+        raise ValueError(f'Unknown Unit {unit} on Axis {axis}')
 
-        self.get_rate()
+    @property
+    @cache
+    def rate(self):
+        r'''
+        Return a rate based on a label in the SVG file.
+        '''
+        rates = self.svgplot.svg.get_texts('(?:scan rate|rate): (?P<value>-?[0-9.]+) *(?P<unit>.*)')
+        # To Do
+        # assert that only one label contains the scan rate
+        # asstert that a rate is available at all
 
-        self.modify_df()
+        # Convert to astropy unit
+        rates[0].unit = CV.get_correct_unit(rates[0].unit)
+        # Convert to SI astropy unit: V / s
+        rate = float(rates[0].value) * rates[0].unit
 
-    def get_rate(self):  # TODO: probably not required
-        r'''
-        Return rate based on the x coordinate units.
+        return rate
 
-        At the moment we simply use the value.
-        '''
-        self.rate = self.description['scan rate']['value']
-        return self.rate
+    @property
+    @cache
+    def cv_df(self):
```

Comment:
Please add a docstring here. It is not clear what `cv_df` is otherwise.

Reply:
Maybe just `dataframe(self)` or `df(self)` instead?

Reply:
I hope my explanation is sufficient